### PR TITLE
adapt to sumk mesh changes in dfttools

### PR DESCRIPTION
**Pull request by the-hampel**
- initialize mesh first in dmft_cycle
- use sum_k.mesh in all instances
- remove iw_or_w

### CODE REVIEW

File: python/solid_dmft/dmft_tools/initial_self_energies.py [hunk 1]
```diff
@@ -306,7 +305,7 @@ def _set_loaded_sigma(sum_k, loaded_sigma, loaded_dc_imp, general_params):
                      for channel in loaded_dc_shell}
                     for calc_dc_shell, loaded_dc_shell in zip(sum_k.dc_imp, loaded_dc_imp)]
     freq = 'w' if general_params['solver_type'] in ['ftps'] else 'iw'
```
Comment (phibeck) [inline python/solid_dmft/dmft_tools/initial_self_energies.py]: This can probably be removed.

**Review by phibeck — CHANGES_REQUESTED** [review]
Please check my one comment, otherwise it looks good to me.

File: python/solid_dmft/dmft_tools/initial_self_energies.py (hunk 1 above)
Comment (the-hampel) [reply python/solid_dmft/dmft_tools/initial_self_energies.py]: fixed in [4ae8e81](https://github.com/TRIQS/solid_dmft/pull/14/commits/4ae8e81bda197b24cf35273606df9b6fff92879e)

**Review by phibeck — COMMENTED** [review]
Looks good, thanks.

**Review by phibeck — APPROVED** [review]

**Pull request merged**